docs/operator-manual/resource_actions.md

```diff
@@ -0,0 +1,60 @@
+# Resource Actions
+
+## Overview
+Argo CD allows operators to define custom actions users can perform on specific resource types. This is used internally to provide actions like `restart` for a `DaemonSet`, or `retry` for an Argo Rollout.
+
+Operators can add actions to custom resources in form of a Lua script and expand those capabilities.
+
+## Custom Resource Actions
+
+Argo CD supports custom resource actions written in [Lua](https://www.lua.org/). This is useful if you:
+
+    * Have a custom resource for which Argo CD does not provide any built-in actions.
+    * Have a commonly performed manual task that might be error prone if executed by users via `kubectl`
+
+
+You can define your own custom resource actions in the `argocd-cm` ConfigMap.
+
+### Define a Custom Resource Action in `argocd-cm` ConfigMap
+
+Custom resource actions can be defined in `resource.customizations` field of `argocd-cm`. Following example demonstrates a set of custom actions for `CronJob` resources. The customizations key is in the format of `apiGroup/Kind`.
+
+```yaml
+resource.customizations: |
+  batch/CronJob:
+    actions: |
+      discovery.lua: |
+        actions = {}
+        actions["suspend"] = {["disabled"] = true}
+        actions["resume"] = {["disabled"] = true}
+
+        local suspend = false
+        if obj.spec.suspend ~= nil then
+            suspend = obj.spec.suspend
+        end
+        if suspend then
+            actions["resume"]["disabled"] = false
+        else
+            actions["suspend"]["disabled"] = false
+        end
+
+        return actions
+      definitions:
+      - name: suspend
+        action.lua: |
+          obj.spec.suspend = true
```

Comment:
Would you be able to help us with a more realistic example, please?

Reply:
I think this example is very realistic. It makes it really easy to suspend and resume CronJobs from UI. I would even bundle it into Argo CD .